public/app/core/utils/explore.ts

```diff
@@ -103,17 +98,16 @@ export function buildQueryTransaction(
   queries: DataQuery[],
   queryOptions: QueryOptions,
   range: TimeRange,
-  queryIntervals: QueryIntervals,
   scanning: boolean
 ): QueryTransaction {
-  const { interval, intervalMs } = queryIntervals;
-
   const configuredQueries = queries.map(query => ({ ...query, ...queryOptions }));
   const key = queries.reduce((combinedKey, query) => {
     combinedKey += query.key;
     return combinedKey;
   }, '');
 
+  const { interval, intervalMs } = getIntervals(range, queryOptions.minInterval, queryOptions.maxDataPoints);
```

Comment:
Maybe we should add a test for this so it doesn't get lost again

Reply:
added unit tests in https://github.com/grafana/grafana/pull/19107/commits/b6cc4d0d562b9bcb0a62dead92dd3f0ec546b684